### PR TITLE
Mhs/uninstallpackage

### PR DESCRIPTION
**Pull request by MartinHS**
- Added list installed packages command and api. 
- Added e2e tests for upcoming features that is meant to be run against a local running instance of humio
- Added file filtering when creating package archive for the install command to exclude unwanted local system specific files

### CODE REVIEW

File: api/packages.go [hunk 1]
```diff
@@ -32,6 +32,11 @@ func (resp *ValidationResponse) IsValid() bool {
 	return (len(resp.InstallationErrors) == 0) && (len(resp.ParseErrors) == 0)
 }
 
+// InstalledPackage contain the details of an installed package
+type InstalledPackage struct {
```
Comment (danielamkaer) [inline api/packages.go:36]: Please add more fields to the struct. This is part of the API package and I would think that users of the API package would like more info than an ID only.

File: api/packages.go [hunk 2]
```diff
@@ -71,6 +76,28 @@ func (p *Packages) Validate(repoOrViewName string, absDiretoryPath string) (*Val
...
+	}
+
+	graphqlErr := p.client.Query(&q, variables)
+
+	var installedPackages []InstalledPackage
+	if graphqlErr == nil {
+		installedPackages = q.Repository.InstalledPackages
+	}
+
+	return installedPackages, graphqlErr
```
Comment (danielamkaer) [inline api/packages.go:98]: Generally acceptable to just do `return q.Repository.InstalledPackages, graphqlErr` here

File: cmd/humioctl/packages_listInstalled.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	"fmt"
+
+	"github.com/humio/cli/prompt"
+
+	"github.com/spf13/cobra"
+)
+
+func listInstalledPackagesCmd() *cobra.Command {
+	cmd := cobra.Command{
+		Use:   "listInstalled [flags] <view-or-repo-name>",
```
Comment (danielamkaer) [inline cmd/humioctl/packages_listInstalled.go:27]: consider `Use: "list-installed` ...

File: cmd/humioctl/packages_listInstalled.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+			out.Info(fmt.Sprintf("Listing installed packages in view %s", viewName))
+
+			// Get the HTTP client
+			client := NewApiClient(cmd)
+
+			installedPackages, err := client.Packages().ListInstalled(viewName)
+
+			if err != nil {
+				return fmt.Errorf("Error fetching parsers: %s", err)
```
Comment (danielamkaer) [inline cmd/humioctl/packages_listInstalled.go:42]: `s/parsers/packages/`

File: cmd/humioctl/packages_listInstalled.go (hunk 3 above)
Comment (danielamkaer) [reply cmd/humioctl/packages_listInstalled.go:27]: also rename the file to `packages_list_installed.go` then.

File: cmd/humioctl/packages_listInstalled.go (hunk 4 above)
Comment (anagrius) [reply cmd/humioctl/packages_listInstalled.go:42]:
```suggestion
				return fmt.Errorf("Error fetching packages: %s", err)
```

File: api/packages.go [hunk 5]
```diff
@@ -71,6 +76,28 @@ func (p *Packages) Validate(repoOrViewName string, absDiretoryPath string) (*Val
 	return &report, nil
 }
 
+// ListInstalled returns a list of installed packages
+func (p *Packages) ListInstalled(repoOrViewName string) ([]InstalledPackage, error) {
```
[the remark this answers is not in the document]
Comment (anagrius) [reply api/packages.go:80]: You are actually not right Daniel. The terminology is horrible, but is as it views and repositories.
Dataspaces is an internal term, not an external one. This is correct.

File: api/packages.go (hunk 1 above)
Comment (anagrius) [reply api/packages.go:36]: I don't agree. Think this is a case of "You ain't gonna need it."
For now we don't need anything else and until we have a way to pass a selection set, this is appropriate.

File: api/packages.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (danielamkaer) [reply api/packages.go:80]: Sure. We should find a fix for the terminology then 😉

File: api/packages.go (hunk 1 above)
Comment (danielamkaer) [reply api/packages.go:36]: Alright. I see also that `ID` in this case is `foo/bar@1.2.3` not a random string.

Comment (MartinHS) [reply api/packages.go:36]: I added a few more fields just because I didn't see this comment before I was done.

**Review by anagrius — APPROVED** [review]

**Pull request merged**